sdk/program/src/message/sanitized.rs

```diff
@@ -433,6 +433,18 @@ pub struct TransactionSignatureDetails {
 }
 
 impl TransactionSignatureDetails {
+    pub fn new(
```

Comment:
this doesn't **strictly** need to be part of this PR any more.
but it is necessary if we want to take the precompile details and turn it into the full details